sdk/provisioning/Azure.Provisioning.Storage/src/BlobService.cs

```diff
@@ -19,11 +21,20 @@ public partial class BlobService
     private partial BicepValue<string> GetNameDefaultValue() =>
         new StringLiteral("default");
 
+    /// <inheritdoc/>
+    IEnumerable<BicepOutput> IClientCreator.GetOutputs()
+    {
+        yield return new BicepOutput($"{ResourceName}_endpoint", typeof(string))
```

Comment:
would this output always be hard coded, or will it eventually dynamically enumerate all available outputs?

Reply:
There are a couple of ways to interpret that question. :smile:
- Should we allow flexible naming schemes?  _Maybe_.   So far I've been thinking about these as "bicep noise" people wouldn't concern themselves with so `RESOURCE_property` seemed fine.  If you work hard to generate pristine bicep and these stand out like a sore thumb, I could see folks wanting something more flexible here.  I'm planning to wait for feedback, but think it'd be pretty cleanly additive.
- Should we be looking for an existing output with that name/type?  **Yes**.  This should eventually behave more like https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/provisioning/Azure.Provisioning/src/Primitives/LocationPropertyResolver.cs#L54.  Then customers who added their own (or had infra that added one) wouldn't be making a mess.  I could probably do that in the InfraResolver today, but I've been kind of punting until we have better namescoping support (i.e., infra knows all of the the named things inside of it for easier access and can help generating unique names)
- Should we be hardcoding `endpoint` here?  No.  This should eventually move into the codegen spec file so we can declare it for any library.  This is one of those hard problems though because ideally we'd generate something that could connect to any language's library since you might declare your infrastructure in C# but use it from JS.  (That's assuming you're just using `IClientCreator` to have config dumped out and saved to disk with future helpers in `ProvisioningDeployment` instead of creating live clients immediately as I like showing in demos.)